app/models/order.rb

```diff
@@ -20,6 +20,10 @@ class Order < ApplicationRecord
   has_many :payments, dependent: :destroy
   has_many :ingredients_meals, through: :meals
 
+  ### CALLBACKS ###
+
+  before_save :check_placed_on, if: :cancelled?
```

Comment:
this callback will run every time once cancelled

Reply:
Done